app/src/main/java/com/example/hungrywolfes/ui/searchScreen/SearchScreenFragment.kt

```diff
@@ -0,0 +1,121 @@
+package com.example.hungrywolfes.ui.searchScreen
+
+import android.app.Activity
+import android.content.Context
+import android.os.Bundle
+import android.util.Log
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.inputmethod.InputMethodManager
+import android.widget.SearchView
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.RecyclerView
+import com.example.hungrywolfes.databinding.FragmentSearchScreenBinding
+import com.example.hungrywolfes.R
+import com.example.hungrywolfes.ui.SearchedImagesAdapter
+
+private const val TAG = "SearchScreenFragment"
+
+class SearchScreenFragment : Fragment() {
+
+    private lateinit var binding: FragmentSearchScreenBinding
+    private val viewModel: SearchViewModel by viewModels()
+    private val searchedImagesAdapter = SearchedImagesAdapter()
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentSearchScreenBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.viewModel = viewModel
+        binding.results.text = getNumberOfResults()
+        setupRecyclerViews()
+        searchImages()
+        setupObservers()
+    }
+
+    private fun searchImages() {
+        binding.searchMeal.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(p0: String?): Boolean {
+                hideKeyboard(context)
+                return true
+            }
+
+            override fun onQueryTextChange(p0: String?): Boolean {
+                if (p0 != null) {
+                    viewModel.getSearchedMeal(p0)
+                }
+                return false
+            }
+        })
+    }
+
+    override fun onResume() {
+        super.onResume()
+        showKeyboard()
+    }
+
+    private fun setupRecyclerViews() {
+        val verticalDivider = DividerItemDecoration(requireContext(), RecyclerView.VERTICAL)
+        val horizontalDivider = DividerItemDecoration(requireContext(), RecyclerView.HORIZONTAL)
+
+        getContextCompat(verticalDivider)
```

Comment:
It is a bit odd to name it get**ContextCompat**. It would also be odd to name it getDrawable since you are not returning any value. Since you are actually **setting** a **drawable** on the **divider** used as parameter, it would make more sense to name it setDividerDrawable or something similar.